python/traverse.py

```diff
@@ -0,0 +1,169 @@
+import timeit
+from collections import deque
+from typing import Callable, Deque, List, Optional, Tuple, Union
+
+import lxml.html
+
+ElementReplacement = Optional[
+    Union[str, lxml.html.HtmlElement, List[lxml.html.HtmlElement]]
+]
+
+# https://developer.mozilla.org/en-US/docs/Glossary/Void_element
+VOID_ELEMENTS = {
+    "area",
+    "base",
+    "br",
+    "col",
+    "embed",
+    "hr",
+    "img",
+    "input",
+    "link",
+    "meta",
+    "source",
+    "track",
+    "wbr",
+}
+
+
+def traverse_and_execute(
+    html: str, fn: Callable[[lxml.html.HtmlElement], None]
+) -> None:
+    elements = lxml.html.fragments_fromstring(html)
+
+    # If there's leading text, the first element of the array will be a string.
+    # We can just discard that.
+    if isinstance(elements[0], str):
+        del elements[0]
+
+    for element in elements:
```

Comment:
Have to `from itertools import chain` at the top.
```suggestion
    for e in chain.from_iterable(element.iter() for element in element_list)
```

Reply:
Done in 8e121e4.